apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/page.tsx

```diff
@@ -162,13 +162,21 @@ export default async function APIKeyDetailPage(props: {
         >
           <ArrowLeft className="w-4 h-4" /> Back to API Keys listing
         </Link>
-        <Link
-          href={`/apis/${props.params.apiId}/keys/${props.params.keyAuthId}/${props.params.keyId}/settings`}
-          className={cn(buttonVariants({ variant: "outline" }), "gap-1")}
-        >
-          <Settings2 className="w-4 h-4" />
-          Key settings
-        </Link>
+
+        <div className="flex items-center gap-4">
```

Comment:
I think it would be better suited on the key's settings page, let's add a new card there to allow rerolling

Reply:
ok, I'll move it to there

Reply:
@chronark, I believe I should get rid of the Dialog and show the content in the card, right?

Reply:
I think a modal can still make sense, this is a potentially destructive action and it should be presented as such.

we can move the form to a card on the settings page, but we still need a modal asking for confirmation